cli/azd/pkg/tools/docker/docker.go

```diff
@@ -69,13 +70,14 @@ func (d *docker) Build(
 	buildContext string,
 	tagName string,
 	buildArgs []string,
+	stdOut io.Writer,
```

Comment:
It turns out that at least on macOS with the latest docker from Docker Desktop, the progress output from `docker build` is writen to `stderr` not `stdout` as it previously was.

We should change this so that it's something like:

`buildProgress io.Writer`

And then if non nil set it for both standard out and standard error, I think.  It looks like the behavior of writing to stderr over stdout happened with the mover to BuildKit (which is a a more modern docker thing) but I think we need to support both because the docker Victor had on his Ubuntu box was still writing to stdout).